### DISCUSSION TITLE
[Issue]: Visualization runs laggy. Are there any ways for me to improve the frame speed?

### DISCUSSION THREAD

**Discussion by DiazaOA** [OpenTOPAS/OpenTOPAS#131] [Announcements]
### Contact Details

diazaokadimara@gmail.com

### What was the issue?

Dear Administrators,

I just (successfully) installed TOPAS on my Windows system. I ran some examples' files and the rundemos.csh. I noticed that the visualization (animation) runs slightly slow and laggy. Is this normal/expected for WSL users, or are there any ways I can improve this?

Thank you for your help. 
Kind regards,
Diaza

### Version

4.0

### What Operative System (OS) are you running on?

Windows (WSL)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by tmasilela** [comment]
Greetings Diaza,

I personally have not used TOPAS on WSL but during the initial testing of the installation I don't think we encountered any problems with visualization, unless of course you're simulating a huge number of particles, in which case it is normal that things are laggy.

Could you point to one specific example where this occurs for you and I'll try find a windows system to test on.

All the best,
Thongchai

**Comment by DiazaOA** [comment]
Dear Thongchai,

Thank you very much for responding to my query. I was trying some of the example files. I attached a screenshot of what I ran and a link containing a short 15-second video of how the visualisation looked. As the video will show, it feels very laggy, with some flickering. Is that expected on a Windows system?

Kindly,
Diaza

![WSL Screenshot](https://github.com/user-attachments/assets/b806d9cf-30ea-4297-b0eb-4b5da3f2e174)

Video capture of my screen:
https://streamable.com/g834wc

**Comment by tmasilela** [comment]
From what I can tell the flickering does unfortunately seem to be a WSL specific issue. If the animation/graphic is important to you I would recommend, for the moment, using the Debian/Mac versions of OpenTOPAS in which you'll see that the animation is much smoother.

If you wish to continue using WSL for visualization please have a look at [this](https://github.com/OpenTOPAS/OpenTOPAS/discussions/43) discussion as it might solve your issue.

All the best,
Thongchai
